### ISSUE TITLE
Support for multiple requests simultaneously

### ISSUE THREAD

**Issue by WillTDA** [nomic-ai/gpt4all-ts#17]
I have created my own expressjs/socket.io web UI using this package. However, this package seems to only allow one connection/request for the model at a time. I could instruct my code to create a new instance of the model every time a user connects to my page, but that would be very inefficient memory-wise.

I had the idea of creating a queue system where the users wait for other requests to complete before serving them, but depending on the length of the answers and how many are waiting, users could be waiting around for a long time.

**TL;DR:** Would it be possible to allow the package or model to support more than one request/prompt simultaneously without dramatically increasing RAM consumption?
